### PR TITLE
Cleanup dependency graph, remove NU1605 warnings

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/75
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: Removing NU1605 warning from NuGet.Client's restore graph completely. 
This meant upgrading an analyzers package that generated fresh new warnings for us. 

Some I have addressed, others I have created issues for. 

https://github.com/NuGet/Home/issues/7674
https://github.com/NuGet/Home/issues/7675
https://github.com/NuGet/Home/issues/7676

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: src/NuGet.Clients/NuGet.Console/Console/ConsoleDispatcher.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+\// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (PatoBeltran) [inline src/NuGet.Clients/NuGet.Console/Console/ConsoleDispatcher.cs]: Remove this change

File: test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj [hunk 2]
```diff
@@ -15,7 +15,7 @@
   </ItemGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == '$(NETCoreTargetFramework)' ">
-    <PackageReference Include="System.Diagnostics.TraceSource" Version="4.0.0" />
+    <PackageReference Include="System.Diagnostics.TraceSource" Version="4.3.0" />
```
Comment (PatoBeltran) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: should we have all of this `4.3.0` versions in a variable?

**Review by PatoBeltran — APPROVED** [review]
Overall this looks good, make sure CI is passing before merging.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj (hunk 2 above)
Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: Maybe.

The reason I decided against is that there are occasional 4.5.x ones (very few). I also wasn't sure what to name that potential variable (SystemPackagesVersion I guess)

I can take another look to understand if it makes sense.

Comment (nkolev92) [reply test/NuGet.Core.Tests/NuGet.Commands.Test/NuGet.Commands.Test.csproj]: @PatoBeltran 
I have added a SystemPackagesVersion....all the packages that start with System* are 4.3.0!

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/GlobalSuppressions.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+// This file is used by Code Analysis to maintain SuppressMessage 
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given 
+// a specific target and scoped to a namespace, type, member, etc.
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "VSTHRD110:Observe result of async calls", Justification = "https://github.com/NuGet/Home/issues/7674", Scope = "member", Target = "~M:NuGet.PackageManagement.PowerShellCmdlets.FindPackageCommand.FindPackagesByPSSearchService")]
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/GlobalSuppressions.cs:9]: why are there new warnings?

**Review by jainaashish — APPROVED** [review]
Looks good.

File: src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/GlobalSuppressions.cs (hunk 3 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/GlobalSuppressions.cs:9]: The analyzers got upgraded as well, courtesy of https://github.com/NuGet/Home/issues/6279. 

None of the warning were super critical (some I'd say we should not even fix), but they will be worth taking a stab at.

**Pull request merged**